utils/load_economy.lua

```diff
@@ -27,13 +27,22 @@ function place_flags(eco, plr)
 end
 
 function place_roads(eco, plr)
-   for i = 1, #eco.roads do
-      local command = string.format("road = plr:place_road(map:get_field(%i,%i).immovable, %s ,true)", eco.roads[i].x, eco.roads[i].y, eco.roads[i].dirs)
+   carrier2 = plr.tribe.carriers[2]
+   for i, rd in ipairs(eco.roads) do
+      -- Road type is not dumped
+      -- Waterways are not dumped currently
+      if rd.workers[carrier2] then
+         road_type = "busy"
+      else
+         road_type = "normal"
+      end
```

Comment:
I wonder whether I should change `dump_economy` to dump road type too, so this wouldn't be needed. (and maybe waterways could be included then?) Relying on that would however break the load script for old dumps. Is that OK, being this a new feature for 1.1?

Reply:
Since it's new, breaking old dumps is fine. +1 for including all road types.